.github/workflows/pr.yaml

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2021 ScyllaDB
+
+name: Verify PR
+
+on:
+  pull_request:
+    branches:
+    - '**'
+    # GitHub doesn't allow to trigger directly when editing a milestone yet.
+    # https://github.community/t/feature-request-add-milestone-changes-as-activity-type-to-pull-request/16778/7
+    #
+    # We have to verify PR milestone in pull_request and we can't use do-not-merge/ label
+    # because it needs to react to master branch changes (when master starts being a different version).
+    # Also GH doesn't trigger on labels created with action token.
+    types:
+    - labeled
+    - unlabeled
+    - opened
+    - edited
+    - reopened
+    - synchronize
+
+jobs:
+  title:
+    name: Verify PR title
+    runs-on: ubuntu-latest
+    steps:
+    - name: Verify PR title
+      run: |
+        set -euExo pipefail
+        shopt -s inherit_errexit
+
+        pr_title=$( jq -r '.pull_request.title' "${GITHUB_EVENT_PATH}" )
+        if echo "${pr_title}" | grep -E \
+          -e '^\[WIP\]' \
```

Comment:
what about catching [WIP] in entire title, not only on the beginning?

Reply:
That's more tricky, I suppose there are valid cases like `Revert: [WIP] A Pr` or other edges. Prow uses prefixes as well.